packages/presentation/lib/app/widgets/tabbar_widget.dart

```diff
@@ -58,8 +73,15 @@ class TabBarWidget extends StatelessWidget {
           ),
         ],
         onTap: (tabIconIndex) {
-          if (tabIconIndex == 0) {
-            _pop();
+          if (tabIconIndex != _currentIndex) {
+            currentIndex = tabIconIndex;
+            switch (tabIconIndex) {
+              case 0:
+                widget._goToHomePage();
+                break;
+              case 3:
+                widget._goToLoginPage();
+            }
```

Comment:
1) Let's move this logic into bloc
2) do it via enum. so, u will have the following flow:

u will have a enum for bottom navigation
enum BottomNavigationItemType {
home
...
...
login
}

u can create a static method/extension, which consumes number as parameter and returns enum item (as each enum has its own index

u refactor your code like this

final type = BottomNavigationItemType.fromIndex(index)
switch (type) {
case BottomNavigationItemType.home -> do smth

it seems much prettier